src/extractor/location_dependent_data.cpp

```diff
@@ -0,0 +1,284 @@
+        };
+
+        auto envelop = append_ring_segments(rings[0].GetArray());
+        bounding_boxes.emplace_back(envelop, polygons.size());
+        for (rapidjson::SizeType iring = 1; iring < rings.Size(); ++iring)
+        {
+            append_ring_segments(rings[iring].GetArray());
+        }
+
+        constexpr const std::size_t segments_per_band = 10;
+        constexpr const std::size_t max_bands = 100000;
+        auto num_bands = segments.size() / segments_per_band;
+        if (num_bands < 1)
+        {
+            num_bands = 1;
+        }
+        else if (num_bands > max_bands)
+        {
+            num_bands = max_bands;
+        }
+
+        polygon_bands_t bands(num_bands);
+
+        const auto y_min = envelop.min_corner().y();
+        const auto y_max = envelop.max_corner().y();
+        const auto dy = (y_max - y_min) / num_bands;
+
+        for (const auto &segment : segments)
+        {
+            using coord_t = boost::geometry::traits::coordinate_type<point_t>::type;
+            const std::pair<coord_t, coord_t> mm =
+                std::minmax(segment.first.y(), segment.second.y());
+            const auto band_min = std::min<coord_t>(num_bands - 1, (mm.first - y_min) / dy);
+            const auto band_max = std::min<coord_t>(num_bands, ((mm.second - y_min) / dy) + 1);
+
+            for (auto band = band_min; band < band_max; ++band)
+            {
+                bands[band].push_back(segment);
+            }
+        }
+
+        polygons.emplace_back(std::make_pair(bands, properties_index));
+    };
+
+    for (rapidjson::SizeType ifeature = 0; ifeature < features_array.Size(); ifeature++)
+    {
+        util::validateFeature(features_array[ifeature]);
+        const auto &feature = features_array[ifeature].GetObject();
+        const auto &geometry = feature["geometry"].GetObject();
+        BOOST_ASSERT(geometry.HasMember("type"));
+
+        // Case-sensitive check of type https://tools.ietf.org/html/rfc7946#section-1.4
+        if (std::strcmp(geometry["type"].GetString(), "Polygon") == 0)
+        {
+            // Collect feature properties and store in polygons vector
+            auto properties_index = collect_properties(feature["properties"].GetObject());
+            const auto &coordinates = geometry["coordinates"].GetArray();
+            index_polygon(coordinates, properties_index);
+        }
+        else if (std::strcmp(geometry["type"].GetString(), "MultiPolygon") == 0)
+        {
+            auto properties_index = collect_properties(feature["properties"].GetObject());
+            const auto &polygons = geometry["coordinates"].GetArray();
+            for (rapidjson::SizeType ipolygon = 0; ipolygon < polygons.Size(); ++ipolygon)
+            {
+                index_polygon(polygons[ipolygon].GetArray(), properties_index);
+            }
+        }
+    }
+
+    // Create R-tree for bounding boxes of collected polygons
+    rtree = rtree_t(bounding_boxes);
```

Comment:
Hrm I don't quite understand how this works with multiple input files: It seems to me on every call of this functions the rtree gets overwritten and all previous data is lost?

Reply:
:man_facepalming: will be fixed